### PR TITLE
[Improve][Resource] Batch upload file

### LINKED ISSUE #12455: [Improvement][Resource] Support batch upload of files
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Support batch upload of files

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Zephyrcf**
## Purpose of the pull request

close https://github.com/apache/dolphinscheduler/issues/12455

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by insist777** [comment]
> ## 拉取请求的目的
> ## 简要变更日志
> ## 验证此拉取请求
> 此拉取请求是代码清理，没有任何测试覆盖范围。
> 
> _（或）_
> 
> 此拉取请求已包含在现有测试中，例如_（请描述测试）。_
> 
> （或）
> 
> 此更改添加了测试，可以按以下方式验证：
> 
> （或）
> 
> 如果您的拉取请求包含不兼容的更改，您还应该将其添加到`docs/docs/en/guide/upgrede/incompatible.md`

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ResourcesServiceImpl.java [hunk 1]
```diff
@@ -214,6 +214,28 @@ private String getFullName(String currentDir, String name) {
                 : String.format(FORMAT_S_S, currentDir, name);
     }
 
+    private List<String> getFullNames(String currentDir, Object[] filesNameSet, Map<String, MultipartFile> fileMap) {
```
Comment (Radeity) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ResourcesServiceImpl.java:217]: Suggest to use `String[] filesNameSet`, type is certain.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ResourcesServiceImpl.java [hunk 2]
```diff
@@ -318,6 +340,128 @@ public Result<Object> createResource(User loginUser,
...
+            return result;
+        }
+
+        result = verifyFiles(type, files);
+        if (!result.getCode().equals(Status.SUCCESS.getCode())) {
+            return result;
+        }
+        Map<String, MultipartFile> fileMap = files.stream().collect(Collectors.toMap(key -> key.getOriginalFilename(), value -> value, (isOld, isNew) -> isOld));
+
+        Object[] filesNameSet = fileMap.keySet().toArray();
```
Comment (Radeity) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ResourcesServiceImpl.java:390]: Please change to `String[] filesNameSet = fileMap.keySet().toArray(new String[0]);`

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ResourcesServiceImpl.java [hunk 3]
```diff
@@ -655,6 +799,91 @@ private Result<Object> verifyFile(String name, ResourceType type, MultipartFile
         return result;
     }
 
+    private Result<Object> verifyFiles(ResourceType type, List<MultipartFile> files) {
```
Comment (Radeity) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ResourcesServiceImpl.java:802]: Can we unify `verifyFiles` and `verifyFile`?

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java [hunk 4]
```diff
@@ -24,6 +24,10 @@
 
 public final class Constants {
 
+    public static final String FILE_NAME_RESTRICTED_CONTENT= "file.name.restricted.content";
```
Comment (Radeity) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java:27]: Please add these properties in `common.properties` and docs.

**Review by Radeity — CHANGES_REQUESTED** [review]
For the scenario of batch uploading file, i suggest to limit the size of batch files and split them into multiple requests, sent in order, which can avoid OOM. 

In addition, can we unify upload single file and batch file?

**Pull request closed without merging**